hotrod/services/frontend/Sources/Run/main.swift

```diff
@@ -0,0 +1,144 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Tracing Examples open source project
+//
+// Copyright (c) YEARS Moritz Lang and the Swift Tracing Examples project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import App
+import ArgumentParser
+import Baggage
+import Instrumentation
+import Lifecycle
+import LifecycleNIOCompat
+import Logging
+import Metrics
+import NIO
+import Prometheus
+import Tracing
+
+struct Serve: ParsableCommand {
+    @Option(name: .shortAndLong, help: "The host which to run on")
+    var host = "localhost"
+
+    @Option(name: .shortAndLong, help: "The port which to run on")
+    var port: UInt = 8080
+
+    @Option(name: .shortAndLong, help: "The mininum log level")
+    var logLevel: String = "info"
+
+    func run() throws {
+        LoggingSystem.bootstrap { label in
+            var handler = StreamLogHandler.standardOutput(label: label)
+            if let logLevel = Logger.Level(rawValue: self.logLevel) {
+                handler.logLevel = logLevel
+            }
+            return handler
+        }
+        let logger = Logger(label: "frontend")
+
+        MetricsSystem.bootstrap(PrometheusClient())
+
+        InstrumentationSystem.bootstrap(LoggingTracer(logger: logger))
+
+        let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: System.coreCount)
+        let app = Application(eventLoopGroup: eventLoopGroup, host: self.host, port: self.port, logger: logger)
+
+        let lifecycle = ServiceLifecycle()
+        lifecycle.register(label: "frontend", start: .eventLoopFuture(app.start), shutdown: .none)
+
+        MetricsSystem.collect = { promise in
+            try MetricsSystem.prometheus().collect(into: promise)
+        }
```

Comment:
this seems weird / confusing... where's that from?

one would periodically call the prom.collect() in a service rather than set some collect func

Reply:
What I wanted to achieve here is to abstract away Prometheus from the `App` target, so in theory different Metrics systems could be used in the future by just configuring this differently, without the `App` target knowing about it.

Reply:
Prometheus is very “weird” though no other system needs collecting and polling really. So this feels weird, also setting statics and thread safety etc... 

We can keep it for now but it likely will have to be removed tbh. Example apps should not do too crazy stuff

Reply:
[`Frontend/AdminService`](https://github.com/slashmo/swift-tracing-examples/blob/feature/jaeger-tracer/hotrod/services/frontend/Sources/Admin/AdminService.swift) now exposes Prometheus metrics without this extension.

I might have been carried away here a bit. It's just too much fun to write nice Swift Server apps. 😁 Getting back to integrating Jaeger now 🙂

Reply:
Okey cool :) will look soon.